eng/pipelines/templates/steps/verify-autorest.yml

```diff
@@ -0,0 +1,66 @@
+parameters:
+  - name: ServiceToTest
+    type: string
+    default: ''
+  - name: SDKType
+    type: string
+    default: 'client'
+  - name: ServiceDirectory
+    type: string
+  - name: NodeVersion
+    type: string
+    default: '12.x'
+  - name: PythonVersion
+    type: string
+    default: '3.9'
+  - name: auto_rest_clone_url
+    type: string
+    default: 'https://github.com/Azure/autorest.python.git'
+
+steps:
+  - task: UsePythonVersion@0
+    displayName: Use
+    inputs:
+      versionSpec: ${{ parameters.PythonVersion }}
+
+  - task: NodeTool@0
+    inputs:
+      versionSpec: ${{ parameters.NodeVersion }}
+
+  - script: |
+      npm install -g autorest
+      autorest --help
+    displayName: "Install autorest"
+
+  - script: |
+      python --version
+      python -m pip install pip==20.1
+      pip install -r eng/autorest_req.txt
+      git clone ${{ parameters.auto_rest_clone_url }}
```

Comment:
nit, as I don't believe autorest repo is that big. but can we use a sparse checkout?

```
  - template: /eng/common/pipelines/templates/steps/sparse-checkout.yml
    parameters:
      Paths:
        - "/"
      Repositories:
        - Name: "Azure/autorest.python"
          Commitish: "master"
          WorkingDirectory: "$(Build.SourcesDirectory)/autorest.python"
      SkipDefaultCheckout: true
```

I'm like 99% certain this will work. This will basically grab zero history for the commitsh you ask for. I'm adding path "/" to tell it to grab every file in the repo.

Reply:
I'm testing this locally to confirm that it's good advice. Feel free to ignore this comment for now.

Reply:
Ok I tested this locally.

This will not work for what I'm asking you to do until I update the `sparse-checkout.yml` to NOT grab the folder `eng` by default. It would work if the pattern was simply:

```
        git clone --no-checkout --filter=tree:0 git://github.com/Azure/autorest.python .
        git sparse-checkout init
        git sparse-checkout add /
        git checkout <targetcommit> (master in your case)
```

But because by default it has a 

```
       git sparse-checkout set eng
```

Run before it adds the path, it actually won't work for git repos without an `eng` folder. It just quietly doesn't checkout properly.

Reply:
[PR that will enable this](https://github.com/Azure/azure-sdk-for-python/pull/16667)

Reply:
@scbedd I would set the path to `/*` instead of `/` which will work with `eng`. This is what git sparse-checkout defaults to using anyway before we run `git sparse-checkout set`

Reply:
Ah ok. I'll definitely give it a shot locally.

Reply:
@seankane-msft this works. I updated the comment with the example. I believe it should be a drop in replacement.